modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -221,9 +226,44 @@ private String _buildFragmentHTML(
 		return sb.toString();
 	}
 
-	private String _getAPIURL(ObjectEntry fdsEntryObjectEntry) {
-		Map<String, Object> properties = fdsEntryObjectEntry.getProperties();
+	private String _embedURLParameters(String apiUrl, HttpServletRequest httpServletRequest){
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		Pattern pattern = Pattern.compile("\\{(.*?)\\}", Pattern.CASE_INSENSITIVE);
+		Matcher matcher = pattern.matcher(apiUrl);
+	
+		if(matcher.find()) {
+			String[] ALLOWED_PARAMETERS = {"siteId", "scopeKey", "userId"};
+
+			for (String param : ALLOWED_PARAMETERS){
+				String paramValue = "";
+
+				switch (param) {
+					case "siteId":
+						paramValue = Long.toString(themeDisplay.getScopeGroupId());
+						break;
+					case "scopeKey":
+						paramValue = Long.toString(themeDisplay.getScopeGroupId());
+						break;
+					case "userId":
+						paramValue = Long.toString(themeDisplay.getUserId());
+						break;
+				}
+
+				apiUrl = apiUrl.replace('{' + param + '}', paramValue);
+			}
```

Comment:
And, at this point, might be useful to log something like "unsupported parameter + <param> in API url" so that we can easily debug cases where fragment can't set the value. This is where regexes can help: if we find the pattern after the known substitutions have been made, then there are more that current logic could not handle.

Reply:
Agree. There's already a validation in frontend, so no unsupported parameters will be find here. However, as long as frontend and backend validations should coexists, it will be useful to cover that case.